src/main/webapp/resources/js/pages/projects/redux/fieldsSlice.js

```diff
@@ -24,17 +33,23 @@ export const fetchFieldsForProject = createAsyncThunk(
 export const fieldsSlice = createSlice({
   name: "fields",
   initialState: {
-    fields: undefined,
     selected: [],
     loading: true,
+    requests: {},
   },
   reducers: {},
   extraReducers: {
-    [fetchFieldsForProject.fulfilled]: (state, action) => ({
-      ...state,
-      fields: action.payload,
-      loading: false,
-    }),
+    [fetchFieldsForProject.fulfilled]: (state, { meta, payload }) => {
+      const requests = { ...state.requests };
```

Comment:
This is really awesome! Love how clean it is :)

Reply:
Just wait until the next PR ;) So much simpler than this even.

Reply:
Awesome!